core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -311,13 +311,24 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
               decommissionExecutors(Array((executorId, v._1)), v._2, v._3)
               unknownExecutorsPendingDecommission.invalidate(executorId)
             })
+          // propagate current log level to new executor only if flag is true
+          if (conf.get(EXECUTOR_ALLOW_SYNC_LOG_LEVEL)) {
+            data.executorEndpoint.send(RefreshExecutor(Map("logLevel" -> Utils.getLogLevel)))
+          }
```

Comment:
The actual issue is, with `spark.log.level` in `sc.conf` or `--conf spark.log.level` or `sc.setLogLevel` ,  executor log level can not be changed. it just changes the Driver side  log level alone
To debug issues on the Executor side, only option currently available is to change the executor log level by modifying log4j/log4j2 properties file. This will force the user to make custom log4j/log4j2 files for each application. 
With this PR, the current log level applied through  `spark.log.leve`l in `sc.conf` or `--conf spark.log.level`  or `sc.setLogLevel` will be passed to executors

Reply:
To clarify, I meant this change at this specific location alone - 
* Set log level in SparkContext.conf when initializing and/or when user calls `setLogLevel`
* When executors are initialized, they already pull the conf from driver - and at this point, apply the log level.

If/when user changes log level dynamically, that would still need to be propagated to executors (as executor(s) have already been initialized) - but we dont need to push it each time a new executor comes up: only when level is changed.

Reply:
Conceptually, I agree with @mridulm. The only thing I'm wondering is if it creates some undesired side-effects if the user turns on the "sync" flag and then sets the log level to overwrite the executor configuration.

I don't think this is a real case though.

Reply:
@mridulm , Done, when executors are initialized, applied the log level of the driver in the executor.
@grundprinzip , If the user turns on the "sync", it will override the executor log configuration only if `spark.log.level`  is applied on `SparkContext` or `sc.setLogLevel()` explicitly called with a different log level than default log level configured on log4j2 , hence there won't be a side effect in that scenario.